flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -46,6 +55,15 @@
     public static final InputTypeStrategy CURRENT_WATERMARK =
             new CurrentWatermarkInputTypeStrategy();
 
+    /** Input strategy for {@link BuiltInFunctionDefinitions#JSON_OBJECT}. */
+    public static final InputTypeStrategy JSON_OBJECT =
```

Comment:
(Placed in-line at the BIFD, Spotless will format this to an absolute mess of a pyramid)

Reply:
This formatting made hard for me to understand in first instance what this code was about :smile:

Reply:
Yeah, our formatting in general sometimes produces things that you kind of just have to get used to. I'm really not a fan of how we format code, but it's what we have. Things can usually be made better by splitting it up more, of course. This here looks OK to me now (granted, you probably need to have an idea of what this is doing).